src/Analyzer.Utilities/SolutionChangeAction.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+
+namespace Analyzer.Utilities
+{
+    public class SolutionChangeAction : CodeAction
+    {
+        private readonly Func<CancellationToken, Task<Solution>> _createChangedSolution;
+
+        public SolutionChangeAction(string title, Func<CancellationToken, Task<Solution>> createChangedSolution)
```

Comment:
Can we enforce this to also take an equivalence key, and throw an ArgumentNullException if null value for it is provided?

Reply:
Fixed. Thank you!

---
In reply to: [155102477](https://github.com/dotnet/roslyn-analyzers/pull/1434#discussion_r155102477) [](ancestors = 155102477)